component/core/src/main/java/org/exoplatform/social/core/listeners/GroupSynchronizationSocialProfileListener.java

```
@@ -0,0 +1,164 @@
+        String propertyValue = (String) property.getValue();
+        Group newPropertyNameGroup = createGroup(propertyName, profileGroup, organizationService);
+        if (newPropertyNameGroup != null) {
+          Group newPropertyValueGroup = createGroup(propertyValue, newPropertyNameGroup, organizationService);
+          addUserToGroup(newPropertyValueGroup, user, organizationService);
+        }
+      });
+    }
+  }
+
+  private Group getGroup(String groupId, OrganizationService organizationService) {
+    try {
+      GroupHandler groupHandler = organizationService.getGroupHandler();
+      return groupHandler.findGroupById(groupId);
+    } catch (Exception e) {
+      return null;
+    }
+  }
+
+  private Group createGroup(String groupName, Group parentGroup, OrganizationService organizationService) {
+    try {
+      Group group = getGroup(buildGroupId(parentGroup, groupName), organizationService);
+      if (group != null) {
+        return group;
+      }
+      GroupHandler groupHandler = organizationService.getGroupHandler();
+      Group newGroup = groupHandler.createGroupInstance();
+      newGroup.setGroupName(groupName.toLowerCase());
+      newGroup.setLabel(StringUtils.capitalize(groupName));
+      newGroup.setDescription(groupName + " group");
+      groupHandler.addChild(parentGroup, newGroup, true);
+      return getGroup(buildGroupId(parentGroup, groupName), organizationService);
+    } catch (Exception e) {
+      return null;
+    }
+  }
+
+  private void addUserToGroup(Group group, User user, OrganizationService organizationService) {
+    if (group == null) {
+      return;
+    }
+    try {
+      MembershipType membershipType = organizationService.getMembershipTypeHandler().findMembershipType(MEMBER);
+      organizationService.getMembershipHandler().linkMembership(user, group, membershipType, true);
+    } catch (Exception e) {
+      LOG.error("Error while adding user to Group", e);
```

same remark as previous about error swallow + throwing for caller to be noticed + logging

In addition, there is no information about the user nor the group. logging such an entry doesn't help and is similar to LOG.error(e)